src/libraries/System.Private.CoreLib/src/System/Threading/RegisteredWaitHandle.Windows.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+#if !FEATURE_WASM_THREADS
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("browser")]
+#endif
+    public sealed partial class RegisteredWaitHandle : MarshalByRefObject
+    {
+        private readonly object? _lock;
+        private SafeWaitHandle? _waitHandle;
+        private readonly _ThreadPoolWaitOrTimerCallback? _callbackHelper;
+        private readonly uint _millisecondsTimeout;
+        private readonly int _signedMillisecondsTimeout;
```

Comment:
Both of these shouldn't be necessary. There is probably other similar consolidation that could be done later in a different PR.